inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp

```diff
@@ -708,158 +758,162 @@ FunctionsComparator::Result FunctionsComparator::compare(
-                    if (!names2.empty())
-                        names2 += ", ";
-                    names2 += name;
-                }
-                names2 = "\"" + names2 + "\"";
-                err_log << "Output tensors names " << names1 << " and " << names2 << " are different for nodes: "
-                        << node1->get_friendly_name() << " and " << node2->get_friendly_name() << std::endl;
-            }
-            if (!node1->output(i).get_partial_shape().same_scheme(
-                    node2->output(i).get_partial_shape())) {
-                err_log << "Different shape detected\n"
-                        << name(node1) << " Output(" << i << ") "
-                        << node1->output(i).get_partial_shape() << " and " << name(node2)
-                        << " Output(" << i << ") " << node2->output(i).get_partial_shape()
-                        << std::endl;
-            }
+    for (int i = 0; i < node1->outputs().size(); ++i) {
+        const auto& tensor1 = node1->output(i).get_tensor();
+        const auto& tensor2 = node2->output(i).get_tensor();
+
+        if (tensor1.get_names() != tensor2.get_names()) {
+            err_log << "Output tensors names " << tensor_names(tensor1) << " and "
+                    << tensor_names(tensor2)
+                    << " are different for nodes: " << node1->get_friendly_name() << " and "
+                    << node2->get_friendly_name() << std::endl;
         }
 
-        if (should_compare(ATTRIBUTES)) {
-            CompareNodesAttributes compare_nodes;
-            node1->visit_attributes(compare_nodes.get_ref_reder());
-            node2->visit_attributes(compare_nodes.get_cmp_reader());
-            if (!compare_nodes.equal()) {
-                return error(
-                    "Comparison of attributes failed for nodes " + name(node1) + ", " +
-                    name(node2) + " [cmp status: " + to_string(compare_nodes) + "]");
-            }
+        if (!node1->output(i).get_partial_shape().same_scheme(
+                node2->output(i).get_partial_shape())) {
+            err_log << "Different shape detected\n"
+                    << name(node1) << " Output(" << i << ") "
+                    << node1->output(i).get_partial_shape() << " and " << name(node2) << " Output("
+                    << i << ") " << node2->output(i).get_partial_shape() << std::endl;
+        }
+    }
+
+    if (should_compare(CmpValues::ATTRIBUTES)) {
+        CompareNodesAttributes compare_nodes;
+        node1->visit_attributes(compare_nodes.get_ref_reader());
+        node2->visit_attributes(compare_nodes.get_cmp_reader());
+        if (!compare_nodes.equal()) {
+            return Result::error(
+                "Comparison of attributes failed for nodes " + name(node1) + ", " + name(node2) +
+                " [cmp status: " + to_string(compare_nodes) + "]");
+        }
+    }
+
+    return Result::ok("Check if any minor error was log in to err_log");
```

Comment:
Non obvious result. It came from minor and major error which might happen during whole `Function` traversing.